perun-base/src/main/java/cz/metacentrum/perun/taskslib/model/Task.java

```diff
@@ -129,54 +147,102 @@ public int getFacilityId() {
 		}
 	}
 
-	public Date getGenEndTime() {
+	public LocalDateTime getGenEndTime() {
 		return genEndTime;
 	}
 
-	public void setGenEndTime(Date genEndTime) {
+	public Long getGenEndTimeAsLong() {
+		return (genEndTime == null) ? null : genEndTime.atZone(ZoneId.systemDefault()).toEpochSecond();
```

Comment:
We need this to return epoch in milliseconds (java Date interpretation) and not unix epoch (in seconds), so GUI will interpret this correctly. Try it like this: ``genEndTime.atZone(ZoneId.systemDefault()).toInstant().toEpochMilli();`` with all getters.